packages/flutter/lib/src/widgets/router.dart

```diff
@@ -967,6 +967,10 @@ class ChildBackButtonDispatcher extends BackButtonDispatcher {
 
   @override
   void takePriority() {
+    if (parent is ChildBackButtonDispatcher) {
+      final ChildBackButtonDispatcher cascadingParent = parent as ChildBackButtonDispatcher;
+      cascadingParent.parent.deferTo(cascadingParent);
```

Comment:
What if there's another ancestor between the root and this parent?

Reply:
oh you are right, it make more sense to put this in defer to.